presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -326,26 +335,51 @@ private PrestoSparkSplitAssigner createSplitAssigner(
         return PrestoSparkPartitionedSplitAssigner.create(session, tableScanNodeId, splitSource, fragmentPartitioning, partitioningProviderManager);
     }
 
-    private ListMultimap<Integer, SerializedPrestoSparkTaskSource> createTaskSources(PlanNodeId tableScanId, SetMultimap<Integer, ScheduledSplit> assignedSplits)
+    private ListMultimap<Integer, SerializedPrestoSparkTaskSource> createTaskSources(PlanNodeId sourceNodeId, SetMultimap<Integer, ScheduledSplit> assignedSplits)
     {
         ListMultimap<Integer, SerializedPrestoSparkTaskSource> result = ArrayListMultimap.create();
         for (int partitionId : ImmutableSet.copyOf(assignedSplits.keySet())) {
             // remove the entry from the collection to let GC reclaim the memory
             Set<ScheduledSplit> splits = assignedSplits.removeAll(partitionId);
-            TaskSource taskSource = new TaskSource(tableScanId, splits, true);
+            TaskSource taskSource = new TaskSource(sourceNodeId, splits, true);
             SerializedPrestoSparkTaskSource serializedTaskSource = new SerializedPrestoSparkTaskSource(serializeZstdCompressed(taskSourceCodec, taskSource));
             result.put(partitionId, serializedTaskSource);
         }
         return result;
     }
 
-    private static List<TableScanNode> findTableScanNodes(PlanNode node)
+    private static <T extends PlanNode> List<T> findAllPlanNodes(PlanNode node, Class<?> clazz)
     {
         return searchFrom(node)
-                .where(TableScanNode.class::isInstance)
+                .where(clazz::isInstance)
                 .findAll();
     }
 
+    private static <T extends PlanNode> Optional<T> findFirstPlanNode(PlanNode node, Class<?> clazz)
+    {
+        return searchFrom(node)
+                .where(clazz::isInstance)
+                .findFirst();
+    }
+
+    private static List<TableScanNode> findTableScanNodes(PlanNode node)
```

Comment:
In general methods should be below the method that calls them.  But see comment on findAllTableScanNodes -- i think we can inline this logic.